### DISCUSSION TITLE
cRexx oo syntax for method invocation

### DISCUSSION THREAD

**Discussion by rvjansen** [adesutherland/CREXX#247] [General]
This subject has quite some history - the Dallas Draft of a new Rexx ANSI standard was mainly about this. To be very short:

1) can we support ~ and . as method call/message pass operator conventions? The . (dot) was problematic in Object Rexx because of the need to support stem. notation. NetRexx opted to drop the dot for stems, instead choosing bracket notation for associative arrays['index']. Object Rexx could not do that - but we can, because if you want stem notation, you can choose level C, while in the other levels we can drop stems. notation and only have arrays[].

This could be building a bridge. I don't know about all the ::requires and ~~ notations. Also, I see no harm in building a new Object Oriented Rexx variant; but there have been more and they all vanished. So the discussion should be about what would unduly fragment the language and what not.

**Comment by adesutherland** [comment]
My view is the default notation will be around the "." and I don't see a problem with this being the same as a stem once you treat a stem as just an object.

That said - it is very easy to add additional support for [] and ~ (which I hate, actually), and indeed we have room to consider if we should use the ":" 

A
